gateway-util-common/pom.xml

```diff
@@ -1,96 +1,123 @@
-    <name>gateway-util-common</name>
-    <description>A collection of common utilities.</description>
+	<dependencies>
+		<dependency>
+			<groupId>org.apache.knox</groupId>
+			<artifactId>gateway-i18n</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>commons-codec</groupId>
+			<artifactId>commons-codec</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>javax.servlet</groupId>
+			<artifactId>javax.servlet-api</artifactId>
+		</dependency>
 
-    <dependencies>
-        <dependency>
-            <groupId>org.apache.knox</groupId>
-            <artifactId>gateway-i18n</artifactId>
-        </dependency>
+		<dependency>
+			<groupId>com.sun.activation</groupId>
+			<artifactId>javax.activation</artifactId>
+		</dependency>
 
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-core</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-databind</artifactId>
-        </dependency>
-        
-        <dependency>
-            <groupId>commons-codec</groupId>
-            <artifactId>commons-codec</artifactId>
-        </dependency>
+		<dependency>
+			<groupId>org.apache.directory.server</groupId>
+			<artifactId>apacheds-jdbm</artifactId>
+		</dependency>
 
-        <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-        </dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-log4j12</artifactId>
+		</dependency>
 
-        <dependency>
-            <groupId>javax.servlet</groupId>
-            <artifactId>javax.servlet-api</artifactId>
-        </dependency>
+		<dependency>
+			<groupId>log4j</groupId>
+			<artifactId>log4j</artifactId>
+		</dependency>
 
-        <dependency>
-            <groupId>com.sun.activation</groupId>
-            <artifactId>javax.activation</artifactId>
-        </dependency>
+		<!-- ********** ********** ********** ********** ********** ********** -->
+		<!-- ********** Test Dependencies ********** -->
+		<!-- ********** ********** ********** ********** ********** ********** -->
 
-        <dependency>
-            <groupId>org.apache.directory.server</groupId>
-            <artifactId>apacheds-jdbm</artifactId>
-        </dependency>
+		<dependency>
+			<groupId>org.apache.knox</groupId>
+			<artifactId>gateway-test-utils</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<!-- <dependency> <groupId>com.auth0</groupId> <artifactId>jwks-rsa</artifactId> 
```

Comment:
Is there a reason the existing `nimbus-jose-jwt` dependency couldn't be used? I looked through the nimbus does and looks like it does JWT verification. I don't think we need to pull in from okta if we can avoid it. Especially not in the gateway-util-common.

Reply:
From a quick look all the changes should be in `gateway-provider-security-jwt` module and not `gateway-util-common`

Reply:
WIP for nimbus-jose-jwt .. looking if nimbus-jose-jwt can  supports JWKS url  (to get keys dynamically )or not . 

Remove all changes  from gateway-util-common

Reply:
nimbus-jose-jwt doesn't support JWKS Url based key fetch . We have to write custom code to fetch keys with httpclient url and do caching  of keys to avoid performance and also need to do invalidating keys in cache .   All these features automatically comes with Okta  JWT . Please suggest .

Reply:
https://connect2id.com/products/nimbus-jose-jwt/examples/jwk-retrieval

^^^ potentially?

Reply:
Thanks @risdenk This API just fetch  list of Keys , still i see we have to do remaining work like converting each key to RSA publickey and do JWT verify  in  loop . I don't have direct nimbus jose api .. still Digging on it

Reply:
looks like https://connect2id.com/products/nimbus-jose-jwt/examples/validating-jwt-access-tokens promising ... i will update as soon as able to implement it . Thanks @risdenk

Reply:
removed okta specific apis . used nimbus-jose-jwt api

Reply:
@risdenk @smolnar82  is there any code formatter template specific for Knox  that i can import in my eclipse

Reply:
Not currently no. Sorry. The formatting of the code in Knox is definitely all over the map. It would be nice to get a handle on that, but needs to be done separately from a functional change.

Reply:
all fixes are done please review  @risdenk